lib/varip.c

```diff
@@ -0,0 +1,89 @@
+#include <rte_ip.h>
+#include <rte_mbuf.h>
+
+#include "gatekeeper_varip.h"
+
+/* NextHeader field of IPv6 header. */
+
+/* Hop-by-hop option header. */
+#define NEXTHDR_HOP      (0)
+
+/* Routing header. */
+#define NEXTHDR_ROUTING  (43)
+
+/* Fragmentation/reassembly header. */
+#define NEXTHDR_FRAGMENT (44)
+
+/* Authentication header. */
+#define NEXTHDR_AUTH     (51)
+
+/* No next header. */
+#define NEXTHDR_NONE     (59)
+
+/* Destination options header. */
+#define NEXTHDR_DEST     (60)
+
+static inline bool
+ipv6_ext_hdr(uint8_t nexthdr)
+{
+	/* Find out if nexthdr is an extension header or a protocol. */
+	return (nexthdr == NEXTHDR_HOP) ||
+		(nexthdr == NEXTHDR_ROUTING) ||
+		(nexthdr == NEXTHDR_FRAGMENT) ||
+		(nexthdr == NEXTHDR_AUTH) ||
+		(nexthdr == NEXTHDR_NONE) ||
+		(nexthdr == NEXTHDR_DEST);
+}
+
+struct ipv6_opt_hdr {
+	uint8_t nexthdr;
+	uint8_t hdrlen;
+} __attribute__((packed));
+
+#define ipv6_optlen(p) (((p)->hdrlen + 1) << 3)
```

Comment:
This macro is used only once, and its code is easier read in the context where it's used. Please just move the code of this macro to where it's used. Remember to drop unnecessary parentheses.